runtime/drivers/github/repo.go

```diff
@@ -99,10 +101,19 @@ func (c *connection) Delete(ctx context.Context, instID, filePath string) error
 
 // Sync implements drivers.RepoStore.
 func (c *connection) Sync(ctx context.Context, instID string) error {
-	r := retrier.New(retrier.ExponentialBackoff(retryN, retryWait), nil)
-	err := r.Run(func() error { return c.pull(ctx) })
+	r := retrier.New(retrier.ExponentialBackoff(retryN, retryWait), retrier.BlacklistClassifier{errUnauthorized})
+	err := r.Run(func() error {
+		err := c.pull(ctx)
+		ghinstallationErr := &ghinstallation.HTTPError{}
+		if errors.As(err, &ghinstallationErr) && ghinstallationErr.Response != nil && ghinstallationErr.Response.StatusCode == 404 {
+			return errUnauthorized
+		}
```

Comment:
I thought we should not retry on any 4xx except for rate limit error?

Reply:
I thought we concluded on only failing for unauthorised errors. I can change this.